backend/sidekiq_config.md

```diff
@@ -0,0 +1,130 @@
+##Background Jobs using Sidekiq
+
+This document outlines the configuration required for a Rails application using the Sidekiq gem for background job processing to be deployed on Heroku.
+
+For detailed information, refer to:
+
+
+* [https://github.com/mperham/sidekiq/wiki/Getting-Started]()
+* [https://devcenter.heroku.com/articles/redistogo]()
+* [https://devcenter.heroku.com/articles/scheduler]()
+
+
+###Rails Configuration
+
+Include the sidekiq gem in the ```Gemfile``` and run install bundle.
+
+```
+  gem ‘sidekiq’
+```
+
+Add the following two lines in the ```app/Procfile```.
+
+```
+  web: bundle exec puma -p $PORT -C ./config/puma.rb
+  worker: bundle exec sidekiq -C config/sidekiq.yml
+```
+
+Add the configuration line for sidekiq in the
+```config/application.rb```.
+
+```
+module AppName
+  class Application < Rails::Application
+    ...
+    config.active_job.queue_adapter = :sidekiq
+  end
+end
+```
+
+Add a sidekiq initializer: ```config/initializers/sidekiq.rb```
+
+```
+# config/initializers/sidekiq.rb
+#
+# Perform Sidekiq jobs immediately in development,
+# so you don't have to run a separate process.
+# You'll also benefit from code reloading.
+if Rails.env.development?
+  require 'sidekiq/testing'
+  Sidekiq::Testing.inline!
+end
+```
+
+Add a sidekiq yml file: ```config/sidekiq.yml```
+
+```
+---
+:verbose: true
+:namespace: sidekiq
+:concurrency: <%= ENV.fetch('SIDEKIQ_CONCURRENCY', 25) %> (3 for Heroku)
+# Set timeout to 8 on Heroku, longer if you manage your own systems.
+:timeout: <%= ENV.fetch('SIDEKIQ_TIMEOUT', 8) %>
+:queues:
+  - [critical, 4]
+  - [high, 3]
+  - [default, 2]
+  - [low, 1]
+```
+Add an environment value in the ```.env file```.
+
+```
+SIDEKIQ_CONCURRENCY=3
+SIDEKIQ_TIMEOUT=8
+```
+
+> If using a free Heroku dyno, be sure to set the Sidekiq concurrency to 3.
+
+###Rails Jobs and Rake Tasks
```

Comment:
I'd add a note here:

We use two general flavors of background jobs: 

(a) those that run after a particular client action (e.g., sending an email on user sign up, send a push notification), and 
(b) those that we want to run at specified intervals (e.g. that pull data from other APIs) 

For the latter, create a rake task that runs your job and schedule it to run at the necessary interval on Heroku using either the Heroku Scheduler addon or, if you want more fine-grained control, the Clockwork addon.